src/display/api.js

```diff
@@ -1742,7 +1752,12 @@ var WorkerTransport = (function WorkerTransportClosure() {
             var size = width * height;
             var rgbaLength = size * 4;
             var buf = new Uint8Array(size * components);
-            var tmpCanvas = createScratchCanvas(width, height);
+            // make sure that this code is not executing in node.js, as
+            // it's using DOM image, and there is no library to support that.
+            assert(typeof document !== 'undefined');
```

Comment:
Either way works for me - it's just to make it easier to catch a problem. We can even move assert/reject at the top of the function.